src/Infrastructure.VS/VsInfo/VsInfoProvider.cs

```diff
@@ -75,5 +78,19 @@ private static IVsVersion CalculateVersion(IVsShell vsShell, ISetupConfiguration
                 return null;
             }
         }
+
+        private static string CalculateName(IVsShell vsShell)
+        {
+            const string defaultName = "Microsoft Visual Studio";
+            try
+            {
+                vsShell.GetProperty((int)__VSSPROPID5.VSSPROPID_AppBrandName, out var name);
+                return name as string ?? defaultName;
+            }
+            catch (Exception ex) when (!ErrorHandler.IsCriticalException(ex))
```

Comment:
The same should actually be done for the other method

Reply:
CriticalException is only relevant to Com calls